client/src/main/java/io/narayana/lra/client/internal/NarayanaLRAClient.java

```diff
@@ -809,26 +793,24 @@ private void endLRA(URI lra, boolean confirm, String compensator, String userDat
                     .header(NARAYANA_LRA_PARTICIPANT_DATA_HEADER_NAME, userData)
                     .async()
                     .put(Entity.text(""))
-                    .get(END_TIMEOUT, TimeUnit.SECONDS);
+                    .get(END_TIMEOUT*100, TimeUnit.SECONDS);
             } catch (InterruptedException | ExecutionException | TimeoutException e) {
-                throw new WebApplicationException("end LRA client request timed out, try again later",
-                        Response.Status.SERVICE_UNAVAILABLE.getStatusCode());
+                throw new WebApplicationException(Response.status(SERVICE_UNAVAILABLE)
+                        .entity("end LRA client request timed out, try again later")
+                        .build());
             }
 
+            if (response == null) {
+                System.out.printf("XXXXXX TODO%n");
```

Comment:
There is a printf to the console here that should be using the logging framework. And on the line below is it definitely fine to throw the Error?

Reply:
I missed this, I'll push another update.